crates/wash-cli/src/common/registry_cmd.rs

```diff
@@ -30,19 +35,54 @@ pub async fn registry_pull(
     cmd: RegistryPullCommand,
     output_kind: OutputKind,
 ) -> Result<CommandOutput> {
-    let artifact_url = cmd.url.to_ascii_lowercase();
-    let image: Reference = artifact_url.parse()?;
+    let artifact_url = match cmd.url {
+        Some(url) => Ok(url),
+        None => {
+            let project_config = get_config(None, Some(true)).map_err(|e| {
```

Comment:
Nit: Instead of `.map_err` where we call `anyhow!`, you should be able to call `.context`, since the error type is already any anyhow error